src/hoverifier.ts

```diff
@@ -897,6 +922,29 @@ export function createHoverifier<C extends object, D, A>({
                     resetHover()
                 }
             })
+            if (scrollBoundaries && scrollBoundaries.length > 0) {
+                // When the hovered token is in this code view, listen to scroll events,
+                // and reset the hover if the hovered token overlaps with any of the scrollBoundaries.
+                subscription.add(
+                    container.updates
+                        .pipe(
+                            distinctUntilChanged(
+                                ({ hoveredTokenElement: e1 }, { hoveredTokenElement: e2 }) => e1 === e2
+                            ),
+                            switchMap(({ codeViewId: id, hoveredTokenElement }) => {
+                                if (id !== codeViewId || !hoveredTokenElement) {
+                                    return NEVER
+                                }
+                                return scrollEvents.pipe(
+                                    filter(() => scrollBoundaries.some(elementOverlaps(hoveredTokenElement)))
+                                )
+                            })
+                        )
+                        .subscribe(() => {
+                            resetHover()
+                        })
+                )
+            }
```

Comment:
One thought I had was whether this should be in the `hoverify` method vs in the `createHoverifier` function above where all the other logic is. It does seem a little weird that we subscribe to the final Observable of state, which will emit a new hover state, but then in some cases we immediately hide the hover (which is another state emission that undoes the previous).

Reply:
@felixfbecker good point, I did this in c753c88. I think this can be tested in Karma, I'll work on that next.